docs/conventions/loadstages.md

```diff
@@ -1,50 +1,51 @@
 Loading Stages
 ==============
 
-Forge loads your mod in 3 main stages: Pre-Initialization, Initialization, and Post-Initialization, commonly referred to as preInit, init, and postInit.
-There are some other events that are important too, depending on what your mod does.
-Each of these stages occurs at a different point in the loading stage and thus what can be safely done in each stage varies.
-
-!!! note
-
-    Loading stage events can only be used in your `@Mod` class, in methods marked with the `@EventHandler` annotation.
+The Forge loading process has four main phases. All of these events shown are fired on the mod-specific eventbus, *not* the global Forge event bus `MinecraftForge.EVENT_BUS`
+Handlers should be registered either using `@EventBusSubscriber(bus = Bus.MOD)` or in the mod object constructor as follows:
+
+```Java
+@Mod("mymod")
+public class MyMod {
+  public MyMod() {
+    FMLModLoadingContext.get().getModEventBus().registerListener(this::commonSetup);
+  } 
+  
+  private void commonSetup(FMLCommonSetupEvent evt) { ... }
+}
+```
 
 !!! warning
+    All four of the below events are called for all mods in parallel. That is, all mods will concurrently receive common setup, FML will wait for 
+    them all to finish, then all mods will concurrently receive sided setup, and so forth.
+    Mods *must* take care to be thread safe, especially when calling other mods' API's and accessing Vanilla systems, which are not thread safe in general.
```

Comment:
Should add a short mention of, or dummy link to info about, the `DeferredWorkQueue`.